### DISCUSSION TITLE
Relacionamento de dados

### DISCUSSION THREAD

**Discussion by felipecesr** [frontendbr/forum#790] [Perguntas e Respostas]
oi galera, boa noite!
como vocês fazem para relacionar dados de uma api com javascript?
por exemplo: tenho uma rota /orders que retorna um json { id: 1, user: 1, ... } e uma rota /users que retorna { id: 1, nome: "John Doe", ... }.
Até aí tudo bem, armazenei os dados de users e orders na minha store mas queria listar orders em uma tabela e a partir do id de user trazer o nome dele que está em users. Criei uma função que usa Array.filter, mas não sei se é a melhor solução

**Comment by bernardodiasc** [comment]
Array.filter é de boa, é uma solução normal. Voce tambem pode usar essa lib https://github.com/paularmstrong/normalizr pra relacionar os dados da forma como deseja.

**Comment by felipeuntill** [comment]
Se você também é o desenvolvedor da api o ideal seria que ela fosse mais completa e não somente um espelho do banco relacional.

Mandar duas requests em minha modesta opinião é mais custoso que extender a funcionalidade básica da api.

Apis foram feitas para serem práticas acima de tudo...

Referências:
[Rest design - Multiple calls vs returning all data in one call](https://softwareengineering.stackexchange.com/questions/294777/rest-design-multiple-calls-vs-returning-all-data-in-one-call)
[Many small requests vs. few large requests (API Design)](https://softwareengineering.stackexchange.com/questions/274340/many-small-requests-vs-few-large-requests-api-design)
[REST API Design: Multiple calls vs. single call to the API](https://softwareengineering.stackexchange.com/questions/341872/rest-api-design-multiple-calls-vs-single-call-to-the-api)
